src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,242 @@
+        }
+    };
+
+    return (
+        <div
+            className={merge([
+                'tw-flex tw-items-center tw-h-9 tw-gap-2 tw-px-3 tw-transition tw-text-s tw-font-sans tw-relative tw-bg-white dark:tw-bg-transparent tw-border tw-rounded tw-line-strong hover:tw-line-x-strong focus-within:tw-line-xx-strong',
+                FOCUS_WITHIN_STYLE,
+                status ? validationClassMap[status] : '',
+            ])}
+            data-test-id={dataTestId}
+        >
+            {decorator ? (
+                <div
+                    className={merge([
+                        'tw-flex tw-items-center tw-justify-center tw-pl-1',
+                        disabled || readOnly ? 'tw-text-black-60' : 'tw-text-black-80',
+                    ])}
+                    data-test-id={`${dataTestId}-decorator`}
+                >
+                    {decorator}
+                </div>
+            ) : null}
+            <input
+                {...props}
+                id={useMemoizedId(propId)}
+                ref={inputElement}
+                name={dataTestId}
+                type="number"
+                placeholder={placeholder}
+                className={merge([
+                    'tw-w-full tw-grow tw-border-none tw-outline-none tw-bg-transparent tw-hide-input-arrows',
+                    disabled || readOnly
+                        ? 'tw-text-black-40 tw-placeholder-black-30 dark:tw-text-black-30 dark:tw-placeholder-black-40'
+                        : 'tw-text-black tw-placeholder-black-60 dark:tw-text-white',
+                ])}
+                onClick={() => inputElement.current?.focus()}
+                onChange={(event) => handleOnChange(event.currentTarget.value)}
+                onBlur={onBlur}
+                onKeyDown={handleKeyDown}
+                required={required}
+                readOnly={readOnly}
+                disabled={disabled}
+                onFocus={onFocus}
+                size={size}
+                aria-label={ariaLabel}
+                title={title}
+                data-test-id={`${dataTestId}-input`}
+            />
+            {incrementable ? (
+                <>
+                    <button
+                        type="button"
+                        onClick={(event) => {
+                            handleCount(NumberInputIncrement.DECREMENT, 1, event.shiftKey, true);
+                        }}
+                        onMouseDown={(event) =>
+                            event.shiftKey ? null : startIncrement(NumberInputIncrement.DECREMENT)
+                        }
+                        onMouseUp={stopIncrement}
+                        onTouchStart={(event) =>
+                            event.shiftKey ? null : startIncrement(NumberInputIncrement.DECREMENT)
+                        }
+                        onTouchEnd={stopIncrement}
+                        aria-label={`${ariaLabel}-decrement`}
+                        title={`${dataTestId}-decrement`}
```

Comment:
```suggestion
                        aria-label="Decrement value"
                        title="Decrement value"
```
These are read by screen readers/are shown on hover, please make sure to have correct text (applies to everywhere)